### PR TITLE
feat(linter): add fixer for `no-empty`

### PR DESCRIPTION
**Pull request by camc314**
adds a suggestion for `no-empty` eslint lint rule

### CODE REVIEW

**Comment by camc314** [comment]
* **#5276** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5276?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camc314 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_linter/src/rules/eslint/no_empty.rs [hunk 1]
```diff
@@ -49,33 +49,94 @@ impl Rule for NoEmpty {
...
+    let src = ctx.source_text();
+    let finally_start = finally_clause.span.start as usize - 1;
+    let mut start = finally_start;
+
+    let src_chars: Vec<char> = src.chars().collect();
+
+    while start > 0 {
+        if let Some(&ch) = src_chars.get(start) {
+            if !ch.is_whitespace() {
+                if ch == 'y' && start >= 6 {
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/eslint/no_empty.rs:124]: Why 6? Will you make this a constant instead of a magic number?

Comment (camc314) [reply crates/oxc_linter/src/rules/eslint/no_empty.rs]: i don't think my braki was working properly when i wrote this.

the `6` was to make sure that on the below line, we don't try to do an out of bounds access.

regardles i rewrite this block - please take another look

File: crates/oxc_linter/src/rules/eslint/no_empty.rs [hunk 2]
```diff
@@ -125,5 +188,29 @@ fn test() {
...
+            "try { foo() } catch (ex) {throw ex} ",
+            None,
+        ),
+        // we can't fix this because removing the `catch` block would change the semantics of the code
+        ("try { foo() } catch (ex) {}", "try { foo() } catch (ex) {}", None),
+        ("if (foo) {}", "", None),
+        ("while (foo) {}", "", None),
+        ("for (;foo;) {}", "", None),
+        ("switch(foo) {}", "", None),
+        ("switch (foo) { /* empty */ }", "", None),
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/eslint/no_empty.rs:204]: Non-blocking: this should not be deleted

**Review by DonIsaac — APPROVED** [review]

File: crates/oxc_linter/src/rules/eslint/no_empty.rs (hunk 2 above)
Comment (camc314) [reply crates/oxc_linter/src/rules/eslint/no_empty.rs:204]: hmm acaually i think this is correct (despite how strange it is)

https://github.com/eslint/eslint/blob/946ae00457265eb298eb169d6d48ca7ec71b9eef/tests/lib/rules/no-empty.js#L134-L142

eslint reports an error

Comment (camc314) [reply crates/oxc_linter/src/rules/eslint/no_empty.rs:204]: @DonIsaac i'm gonna merge this as is, if you think this is incorrect can you make an issue and assign this to me?

i think it makes sense to fix this in a separate PR as we currently report an error for this (maybe we shouldn't

**Pull request merged**